### PR TITLE
Fix higher order components wrapping user-written components

### PR DESCRIPTION
**Pull request by balazsbajorics**
Fixes parts of #1171

**Problem:**
When wrapping Card into `emotion.styled(Card)()`, the ComponentRendererComponent would throw a `Utopia Error: Instance Path is not provided` message.

**Fix:**
The ComponentRendererComponent tries to fall back to using `data-paths` if it does not find the `data-utopia-instance-path` it is looking for.

**Commit Details:**
- Created `tryToGetInstancePath` in ComponentRendererComponent
- the storyboard root now gets an empty instance path instead of an empty template path

### CODE REVIEW

**Review by enidemi — APPROVED** [review]

File: editor/src/components/canvas/ui-jsx-canvas.tsx
Comment (Rheeseyb) [inline editor/src/components/canvas/ui-jsx-canvas.tsx:461]:
```suggestion
  rootInstancePath: TemplatePath
```

Please do not undo my work :sweat_smile:

**Review by Rheeseyb — COMMENTED** [review]
Please can you hold off on merging this until #1177 has been merged, as it's introducing more `InstancePath`s that can instead be `TemplatePath`s

**Review by seanparsons — APPROVED** [review]

**Pull request merged**
